app/forms/Lrn.scala

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright 2019 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package forms
+
+import play.api.data.Forms._
+import play.api.libs.json.{Format, JsString, Reads, Writes}
+import utils.validators.forms.FieldValidator.{isAlphanumeric, noLongerThan}
+
+case class Lrn(value: String) {
+  def isEmpty: Boolean = value.isEmpty
+  def nonEmpty: Boolean = !isEmpty
+}
+
+object Lrn {
+  implicit val format: Format[Lrn] =
+    Format[Lrn](Reads.StringReads.map(Lrn.apply), Writes[Lrn](lrn => JsString(lrn.value)))
+
+  private val lrnMaxLength = 22
+
+  def mapping(prefix: String) =
```

Comment:
Nit: Just thinking... would it be giving any value to use an Enum for prefixes? It is easy for the client to make a mistake/mistype. 
And the Enum could be even inside `Lrn` object.

Reply:
This is used on level on other level mapping - it complete to user for this API it hard to tell what are good solution here. From my site i followed KISS here.